plugins/out_es/es.c

```diff
@@ -232,6 +232,45 @@ static flb_sds_t es_get_id_value(struct flb_elasticsearch *ctx,
     return tmp_str;
 }
 
+static int compose_index_header(struct flb_elasticsearch *ctx,
+                                int es_index_custom_len,
+                                char *logstash_index, size_t logstash_index_size,
+                                char *separator_str,
+                                struct tm *tm)
+{
+    int ret;
+    int len;
+    char *p;
+    size_t s;
+
+    /* Compose Index header */
+    if (es_index_custom_len > 0) {
+        p = logstash_index + es_index_custom_len;
+    } else {
+        p = logstash_index + flb_sds_len(ctx->logstash_prefix);
+    }
+    len = p - logstash_index;
+    ret = snprintf(p, logstash_index_size - len, "%s",
+                   separator_str);
+    if (ret > logstash_index_size - len) {
+        /* exceed limit */
+        return -1;
+    }
+    p += strlen(separator_str);
+    len += strlen(separator_str);
```

Comment:
what do you think about using the sds_printf or sds_cat interfaces ?

Reply:
https://github.com/fluent/fluent-bit/blob/v2.0.9/plugins/out_es/es.c#L259
We can't use sds API since `logstash_index` is char array.

If we change the data type, there will be more diff lines.
I think it is better to create another PR to change the type for refactoring.

Reply:
ping